pypeit/core/coadd.py

```diff
@@ -1390,179 +1557,135 @@ def ech_coadd(files,objids=None,extract='OPT',flux=True,giantcoadd=False,ordersc
-                plt.setp(ax2.get_xticklabels(), visible=False)
-
-                medf = np.median(spec1d_final.flux)
-                ylim = (np.sort([0. - 0.3 * medf, 5 * medf]))
-                cmap = plt.get_cmap('RdYlBu_r')
-                for idx in range(spectra_coadd_rebin.nspec):
-                    spectra_coadd_rebin.select = idx
-                    color = cmap(float(idx) / spectra_coadd_rebin.nspec)
-                    ind_good = spectra_coadd_rebin.sig > 0
-                    ax1.plot(spectra_coadd_rebin.wavelength[ind_good], spectra_coadd_rebin.flux[ind_good], color=color)
-
-                if (np.max(spec1d_final.wavelength) > (9000.0 * units.AA)):
-                    skytrans_file = resource_filename('pypeit', '/data/skisim/atm_transmission_secz1.5_1.6mm.dat')
-                    skycat = np.genfromtxt(skytrans_file, dtype='float')
-                    scale = 0.85 * ylim[1]
-                    ax2.plot(skycat[:, 0] * 1e4, skycat[:, 1] * scale, 'm-', alpha=0.5)
-
-                ax2.plot(spec1d_final.wavelength, spec1d_final.sig, ls='steps-', color='0.7')
-                ax2.plot(spec1d_final.wavelength, spec1d_final.flux, ls='steps-', color='b')
-
-                ax1.set_xlim([np.min(spec1d_final.wavelength.value), np.max(spec1d_final.wavelength.value)])
-                ax2.set_xlim([np.min(spec1d_final.wavelength.value), np.max(spec1d_final.wavelength.value)])
-                ax1.set_ylim(ylim)
-                ax2.set_ylim(ylim)
-                ax1.set_xlabel('Wavelength (Angstrom)')
-                ax1.set_ylabel('Flux')
-                ax2.set_ylabel('Flux')
-
-                plt.tight_layout(pad=0.2, h_pad=0., w_pad=0.2)
-
-                if len(qafile.split('.')) == 1:
-                    msgs.info("No fomat given for the qafile, save to PDF format.")
-                    qafile = qafile + '.pdf'
-                if qafile:
-                    plt.savefig(qafile)
-                    msgs.info("Wrote coadd QA: {:s}".format(qafile))
-                if debug:
-                    plt.show()
-                plt.close()
-
-            ### Do NOT remove this part althoug it is deprecated.
-            # we may need back to using this pieces of code after fixing the coadd_spectra problem on first order.
-            #kwargs['echelle'] = True
-            #kwargs['wave_grid_min'] = np.min(wave_grid)
-            #kwargs['wave_grid_max'] = np.max(wave_grid)
-            #spec1d_final = coadd_spectra(spectra_coadd_rebin, wave_grid_method=wave_grid_method, niter=niter,
-            #                                  scale_method=scale_method, do_offset=do_offset, sigrej_final=sigrej_final,
-            #                                  do_var_corr=do_var_corr, qafile=qafile, outfile=outfile,
-            #                                  do_cr=do_cr, debug=debug, **kwargs)
+            spec1d_final = merge_order(spectra_coadd_rebin, wave_grid, extract=extract, orderscale=orderscale,
```

Comment:
If you're just going to return the object, there's no need to assign it to something.  I.e., just `return merge_order(...)`.